### PR TITLE
sync-diff-inspector: add descriptions and examples of table routers

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs/pull/15242
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sync-diff-inspector/route-diff.md [hunk 1]
```diff
@@ -61,6 +61,137 @@ target-schema = "test_2"       # 目标库名
...
+如果规则中只存在 `target-schema`，如下所示：
+
+```toml
+[routes.rule1]
+schema-pattern = "schema_2"  # 匹配数据源的库名，支持通配符 "*" 和 "?"
+target-schema = "schema"     # 目标库名
+```
+
+sync-diff-inspector 的行为如下：
+
```
Comment (lyzx2001) [inline sync-diff-inspector/route-diff.md:82]: Seems like we do not need this sentence

**Review by lyzx2001 — COMMENTED** [review]
rest lgtm

**Review by Leavrth — APPROVED** [review]

File: sync-diff-inspector/route-diff.md [hunk 2]
```diff
@@ -61,6 +61,135 @@ target-schema = "test_2"       # 目标库名
...
+    ```toml
+    [routes.rule1]
+    schema-pattern = "schema_2"  # 匹配数据源的库名，支持通配符 "*" 和 "?"
+    target-schema = "schema"     # 目标库名
+    ```
+
+    - 如果上游中不存在库 `schema`，sync-diff-inspector 不做任何处理。
+    - 如果上游中存在库 `schema`，且存在一条规则将该库匹配到其他库，sync-diff-inspector 不做任何处理。
+    - 如果上游中存在库 `schema`，但不存在将该库匹配到其他表的规则，sync-diff-inspector 会在表路由中添加一条新的规则 `schema -> _no__exists__db_`。之后，sync-diff-inspector 会将库 `schema` 视为库 `_no__exists__db_`。
```
Comment (ran-huang) [inline sync-diff-inspector/route-diff.md:83]:
```suggestion
    - 如果上游中存在库 `schema`，但不存在将该库匹配到其他库的规则，sync-diff-inspector 会在表路由中添加一条新的规则 `schema -> _no__exists__db_`。之后，sync-diff-inspector 会将库 `schema` 视为库 `_no__exists__db_`。
```

**Review by ran-huang — APPROVED** [review]

File: sync-diff-inspector/route-diff.md [hunk 3]
```diff
@@ -61,6 +61,135 @@ target-schema = "test_2"       # 目标库名
...
+
+    - 如果存在一条规则将 `schema.table` 匹配到 `schema.table`，sync-diff-inspector 不做任何处理。
+    - 如果不存在将 `schema.table` 匹配到 `schema.table` 的规则，sync-diff-inspector 会在表路由中添加一条新的规则 `schema.table -> _no__exists__db_._no__exists__table_`。之后，sync-diff-inspector 会将表 `schema.table` 视为表 `_no__exists__db_._no__exists__table_`。
+
+- 如果规则中只存在 `target-schema`，如下所示：
+
+    ```toml
+    [routes.rule1]
+    schema-pattern = "schema_2"  # 匹配数据源的库名，支持通配符 "*" 和 "?"
+    target-schema = "schema"     # 目标库名
```
Comment (hfxsd) [inline sync-diff-inspector/route-diff.md:78]:
```suggestion
    target-schema = "schema"     # 目标库名 
```

**Pull request merged**